### DISCUSSION TITLE
Downloader appears to run indefinitely

### DISCUSSION THREAD

**Discussion by beebutler** [nianeyna/ao3downloader#150] [Q&A] [answered]
![Screenshot 2024-08-06 08 18 23](https://github.com/user-attachments/assets/cbbb58b9-9636-49ee-bcac-496bb031f2e7)

Thank you for your time as always. This downloader is amazing!

I know that AO3 has issues with EPUBs as they get bigger in size, so I intentionally ran the downloader on a profile that has a few works but multiple EPUBs that won't download through AO3 (https://archiveofourown.org/users/EstrellaRose/pseuds/EstrellaRose). This is me knowing that you sometimes have to use FanFicFare. I was expecting some sort of error, but I let it run for quite some time, and it just keeps finding new pages even though there are only 10 works on the first page. Is there anything I can do to resolve this?

**Comment by nianeyna** [comment, marked as answer]
Ah, yes - this is a known behavior that I've never quite figured out how to fix. Thankfully, the workaround is easy - just click the "Works" link in the sidebar and download from there, instead of from the user's main page

**Reply by beebutler** [reply]
![Screenshot 2024-08-07 18 58 41](https://github.com/user-attachments/assets/2b82aab6-28a0-4278-a6b7-39cd2e49877c)

Thank you for the reply. Noted. I ran it again and it did download seven of the ten fics. Is there any way to run it in a more verbose mode so I can get an error message or at least see which ones are not downloaded without manually checking? I know this is a rather specific instance, but I anticipate having some fics that will be too big to be downloaded with the downloader. This isn't me trying to waste your time, I promise.

**Reply by nianeyna** [reply]
There's no option to print extra data to the console itself no, but I hopefully have something better for you - use menu option "v" and open the resulting html file (it will be located in the "logs" folder) in a browser.

**Reply by beebutler** [reply]
Alright, I have the log file here and I'm looking at all five columns. It's showing that everything was successfully downloaded for this specific author, but I know there were three failures. I can't upload it here directly for reference. Would you like to see it?

**Reply by beebutler** [reply]
My apologies, I got into work mode and forgot to acknowledge your reply. Thank you, as always. And yes, if it worked, this would be an excellent option.

**Reply by nianeyna** [reply]
Yes, you can email me your log file - the .jsonl file if you can please, rather than the html. My email address is in the readme, at the bottom in the Contact section.

I was unable to reproduce your issue, by the way. On running that author's works page through the downloader, I see 10 log entries and 10 downloaded files. Curious.

**Reply by beebutler** [reply]
Since you said you couldn't reproduce the issue, I installed it in a new folder and ran a clean setup. It worked for me as well, so I'm not sure what happened. I guess I'll do some spot checks.

Wait, let me interject here. Are you saying you haven't had issues with fics over 500,000 words? When I try to download the three that failed directly through AO3, I get an error. Maybe I'm starting from the wrong place. Were you not aware of this issue with AO3 that larger fics as EPUBs will not download? If this is new to you, do you want the logs from the folder where it didn't work correctly?

**Reply by nianeyna** [reply]
Indeed, I think this is the first time I've heard of word count causing a problem with downloads. I've seen people have issues with works that have a lot of embedded images, but not with works that are just long. And yes, I would be interested in seeing that log file.
